src/makers/generic/zip.js

```diff
@@ -4,6 +4,9 @@ import pify from 'pify';
 import zipFolder from 'zip-folder';
 
 import { ensureFile } from '../../util/ensure-output';
+import { checkSupportedPlatforms } from '../../util/check-supported-platforms';
```

Comment:
This `zip` maker zips up the wrong content when cross-compiling. See the `switch` statement below -- it's using `process.platform`, so if that's different from the target platform, it will zip up the wrong content.

To fix this, we would need to pass the target platform into the makers, which certainly makes sense, but would change the maker API. How much of a concern is that? Do we think there are many makers out there other than what is implement inside `electron-forge`? Or would we be okay breaking compatibility with external makers?

My proposal would be to change the makers to switch to an options hash instead of positional arguments:

```
export default async ({ dir, appName, targetPlatform, targetArch, forgeConfig, packageJSON }) => {
```

Then if we really care about compatibility, we could implement a helper:

```
export default function callMaker(maker, options) {
  if (maker.length === 1) {
    return maker(options);
  } else {
    let { dir, appName, targetArch, forgeConfig, packageJSON } = options;
    return maker(dir, appName, targetArch, forgeConfig, packageJSON);
  }
}
```

thoughts?